### PR TITLE
Refactored cache to support the nested map structure we whiteboarded

### PR DESCRIPTION
**Pull request by JakeCooper**
Fun fact: This commit was pushed from 30,000 ft in the air. :airplane:

### CODE REVIEW

File: quotecache/quotecache.go [hunk 1]
```diff
@@ -27,26 +27,36 @@ func (q Quote) IsExpired() bool {
 	return time.Now().After(expiry)
 }
 
+var QuoteCache = make(map[string]map[string]Quote)
```
Comment (trstephen) [inline quotecache/quotecache.go:30]: Do we want to export this?

File: quotecache/quotecache.go [hunk 2]
```diff
@@ -27,26 +27,36 @@ func (q Quote) IsExpired() bool {
...
 // GetQuote : Gets the current value of the stock, hitting the local cache if it can.
 func GetQuote(userID, stock string) (Quote, error) {
 	// check if the value is in cache
-	quote, found := cache[stock]
 
-	if !found || quote.IsExpired() {
+	var userQuote Quote
+	userMap := QuoteCache[stock]
+	userQuote, found := userMap[userID]
+	isFresh := userQuote.Timestamp.Unix() - time.Now().Unix() <= 60
```
Comment (trstephen) [inline quotecache/quotecache.go]: Can use `userQuote.IsExpired()` here.

File: quotecache/quotecache.go (hunk 1 above)
Comment (JakeCooper) [reply quotecache/quotecache.go:30]: Not really. It should stay as a pseudo private element of the quotecache.go file so people don't dig into it from outside.

File: quotecache/quotecache.go (hunk 2 above)
Comment (JakeCooper) [reply quotecache/quotecache.go]: Sure, but why does userQuote.IsExpired() use 45s instead of the 60s? Is that a mistake?

Comment (trstephen) [reply quotecache/quotecache.go]: I was planning to use it as a buffer to account for delay, etc. It's a bad magic number!

We could change that 45 -> 60 or refactor like
```go
func (q Quote) IsExpired(lifetime int) bool
```

**Pull request merged**